netmaster/master/netmaster.go

```diff
@@ -196,12 +214,12 @@ func CreateTenant(stateDriver core.StateDriver, tenant *intent.ConfigTenant) err
 		return err
 	}
 
-	if GetClusterMode() == "docker" {
+	if IsDNSEnabled() {
 		// start skydns container
 		err = startServiceContainer(tenant.Name)
 		if err != nil {
-			log.Errorf("Error starting service container. Err: %v", err)
-			return err
+			log.Errorf("Error starting service container. Err: %v. Disabling DNS option.", err)
+			SetDNSEnabled(false)
```

Comment:
I know we discussed this. But, just realized that we should not globally disable DNS mode when start container fails for one tenant. We should probably keep track of this at per tenant level..